scripts/generate_data.mk

```diff
@@ -0,0 +1,57 @@
+
+## Run all data generators
+##   $ make -B -f scripts/generate_data.mk
+## Run specific data generator
+##   $ make -B -f scripts/generate_data.mk spec/std/string/graphemes_break_spec.cr
+
+ifeq ($(OS),Windows_NT)
+  BIN_CRYSTAL=bin\crystal
+else
+  BIN_CRYSTAL=bin/crystal
+endif
+
+all: ## Run all generators
+.PHONY: all
+
+all: spec/std/string/graphemes_break_spec.cr
+spec/std/string/graphemes_break_spec.cr: scripts/generate_grapheme_break_specs.cr
+	$(BIN_CRYSTAL) run $<
+
+all: src/string/grapheme/properties.cr
+src/string/grapheme/properties.cr: scripts/generate_grapheme_properties.cr
+	$(BIN_CRYSTAL) run $<
+
+all: src/openssl/ssl/defaults.cr
+src/openssl/ssl/defaults.cr: scripts/generate_ssl_server_defaults.cr
+	$(BIN_CRYSTAL) run $<
+
+all: src/unicode/data.cr
+src/unicode/data.cr: scripts/generate_unicode_data.cr
+	$(BIN_CRYSTAL) run $<
+
+all: src/crystal/system/win32/zone_names.cr
+src/crystal/system/win32/zone_names.cr: scripts/generate_windows_zone_names.cr
+	$(BIN_CRYSTAL) run $<
+
+all: src/html/entities.cr
+src/html/entities.cr: scripts/generate_html_entities.cr scripts/html_entities.ecr
+	$(BIN_CRYSTAL) run $<
```

Comment:
@HertzDevil Do you think this would be portable enough?

```suggestion
## Run all data generators
##   $ make -f scripts/generate_data.mk

ifeq ($(OS),Windows_NT)
  BIN_CRYSTAL=bin\crystal
else
  BIN_CRYSTAL=bin/crystal
endif

.PHONY: all
all: ## Run all generators
	$(BIN_CRYSTAL) run scripts/generate_grapheme_break_specs.cr
	$(BIN_CRYSTAL) run scripts/generate_grapheme_properties.cr
	$(BIN_CRYSTAL) run scripts/generate_ssl_server_defaults.cr
	$(BIN_CRYSTAL) run scripts/generate_unicode_data.cr
	$(BIN_CRYSTAL) run scripts/generate_windows_zone_names.cr
	$(BIN_CRYSTAL) run scripts/generate_html_entities.cr
```

Reply:
An alternative to listing every single generator file could be a glob `scripts/generate_*.cr`. We could put this easily in both Makefiles and skip `generate_data.mk`.

Reply:
That works, but when the contents are so short they might as well be inlined into the original Makefiles, whose `generate_data` would have no prereqs at all

Reply:
Yes, of course it could fit there. But this is a chance for deduplication.
I'm fine either way though.

Reply:
I think we have three options to move forward:

1. Trim down `generate_data.mk` from this branch as suggested in https://github.com/crystal-lang/crystal/pull/14015#discussion_r1404207911 (avoids duplication)
2. Drop `generate_data.mk` and put the recipe steps from that suggestion directly into the `generate_data` recipes of `Makefile` and `Makefile.win` (keep duplicated script lists)
3. Use a generic command to run all generator scripts for the `generate_data` recipes of `Makefile` and `Makefile.win`: `ls scripts/generate_*.cr | xargs -n1 bin/crystal run` (one-liner, so duplication isn't relevant)

Reply:
`ls` and `xargs` are not on Windows, so 3 is a no.

I'm fine with 1.